local-dev/bin/render

```diff
@@ -2,26 +2,88 @@
 
 set -eo pipefail
 
-REPO_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/../.." && pwd )"
+##Pull mysql instance info for sqlproxy##
+RAWLS_MYSQL_INSTANCE_JSON=$(gcloud secrets versions access latest  --project broad-dsde-dev --secret rawls-mysql-instance)
 
-IMAGE_NAME="${IMAGE_NAME:-us-central1-docker.pkg.dev/dsp-artifact-registry/firecloud-develop-shim/firecloud-develop-shim}"
-IMAGE_TAG="${IMAGE_TAG:-latest}"
-IMAGE="${IMAGE_NAME}:${IMAGE_TAG}"
+# Extract the name and zone values using jq
+CLOUDSQL_ZONE=$(echo "$RAWLS_MYSQL_INSTANCE_JSON" | jq -r '.zone')
+CLOUDSQL_INSTANCE=$(echo "$RAWLS_MYSQL_INSTANCE_JSON" | jq -r '.name')
 
-VAULT_TOKEN=$( cat ~/.vault-token )
+# Define the template and output file paths
+MYSQL_TEMPLATE_FILE="local-dev/templates/sqlproxy.env"
+MYSQL_OUTPUT_FILE="config/sqlproxy.env"
 
-docker run --pull always \
-  -e VAULT_TOKEN="${VAULT_TOKEN}" \
-  -v "${REPO_ROOT}:/code" \
-  --rm -it "${IMAGE}" \
-  --secrets /code/local-dev/secrets.yaml \
-  --templates /code/local-dev/templates \
-  --output-dir /code/config
+# Replace placeholders in the template file with the fetched secret values
+sed -e "s/{{ \$secret.zone }}/$CLOUDSQL_ZONE/" \
+    -e "s/{{ \$secret.name }}/$CLOUDSQL_INSTANCE/" \
+    "$MYSQL_TEMPLATE_FILE" > "$MYSQL_OUTPUT_FILE"
 
-cat <<EOF
+#Files to copy directly to config##
+cp local-dev/templates/docker-rsync-local-rawls.sh config/docker-rsync-local-rawls.sh
+cp local-dev/templates/liquibase.properties config/liquibase.properties
+cp local-dev/templates/local-mysql.conf config/local-mysql.conf
+cp local-dev/templates/local-sam.conf config/local-sam.conf
+cp local-dev/templates/mod_security_ignore.conf config/mod_security_ignore.conf
+cp local-dev/templates/oauth2.conf config/oauth2.conf
+cp local-dev/templates/site.conf config/site.conf
 
-Don't forget to add the local.broadinstitute.org alias to your hosts file!
+##Files to download from secrets##
+gcloud secrets versions access latest --project broad-dsde-dev --secret rawls-sa > config/rawls-account.json
+gcloud secrets versions access latest --project broad-dsde-dev --secret rawls-sa > config/sqlproxy-service-account.json
+gcloud secrets versions access latest --project broad-dsde-dev --secret rawls-sa-pem | jq -r '."key.pem"' > config/rawls-account.pem
+gcloud secrets versions access latest --project broad-dsde-dev --secret rawls-bigquery-sa > config/bigquery-account.json
+gcloud secrets versions access latest --project broad-dsde-dev --secret rawls-billing-sa | jq -r '."key.pem"' > config/billing-account.pem
+gcloud secrets versions access latest --project broad-dsde-dev --secret buffer-client-sa-b64 | jq -r '.key' | base64 --decode > config/buffer-account.json
```

Comment:
might wanna create a helper function for `gcloud secrets versions access latest --project broad-dsde-dev --secret %1` to avoid all the copy/paste … this would also make a central place to change `broad-dsde-dev` in case anyone ever wanted to read secrets from elsewhere